syngenta_digital_dta/s3/adapter.py

```diff
@@ -238,3 +251,23 @@ def __set_acl(self, public_read):
 
     def __generate_publish_data(self, **kwargs):
         return {'presigned_url': self.create_public_url(**kwargs)}
+
+    def __create_tags(self, **kwargs):
+        tags = kwargs.get('tags', None)
+        result_tags = []
+        try:
```

Comment:
Is it better to check for `None` and the type of the param rather than doing `try-except` ? 

if the user tags param is absent or passed as `None`, then the resulted tags should be an empty list, otherwise, it should be a `dict()`, and if not utility should throw a meaningful exception stating that tags param is invalid